src/execution_plan/ops/op_unwind.c

```diff
@@ -32,36 +37,88 @@ OpBase *NewUnwindOp(uint record_idx, AR_ExpNode **exps) {
 }
 
 OpResult UnwindInit(OpBase *opBase) {
+	OpUnwind *unwind = (OpUnwind *)opBase;
+
+	// check if there are children - if not, add static record
+	if(!unwind->op.childCount)
+		unwind->currentRecord = Record_New(opBase->record_map->record_len);
+
+	// check for modifiers in the AR_EXP for static or dynamic list
+	rax *modifies = raxNew();
```

Comment:
if there are no child operations there couldn't be any modifiers, as such this check is redundant and should be removed.

Reply:
but you can have a child operation and have a static list to unwind.
`MATCH(n) UNWIND [0,1,2] as y return n, y`
i changed the code a bit